scripts/docker/update_dockers_json.py

```diff
@@ -0,0 +1,212 @@
+
+    for name, image in updated_images.items():
+        if name == "name":
+            continue
+
+        c += 1
+        pprint_log(f"[{c}/{keys_count}]\t", Status.NO_MARK, line_break=False)
+        image_base = image.split(":")[0]
+        expected_image = f"{image_base}:{tag}"
+        if expected_image == image:
+            pprint_log(name, Status.NEUTRAL)
+            continue
+
+        exists, error = check_image_exist(expected_image)
+        if exists:
+            updated_images[name] = expected_image
+            pprint_log(name, Status.SUCCESS)
+        else:
+            # There might be authentication issues checking if the image
+            # exist; hence checking for the original image's existence;
+            # if this fails too, then there are issues unrelated to the
+            # given tag, else we can assume an image with the given tag
+            # does not exist (in other words, the image is not updated).
+            e, _ = check_image_exist(image)
+            if not e:
+                pprint_log(f"{name}\t{error}", Status.FAILED)
+            else:
+                pprint_log(name, Status.NEUTRAL)
+                continue
+    return updated_images
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="For a given Docker image tag, this script checks "
+                    "if any of the Docker images listed in the provided JSON "
+                    "file (e.g., `dockers.json`) are pushed to the container "
+                    "registry (e.g., Google Container Registry ) with that "
+                    "tag.\n\n"
+                    "If the given tag is different from the tag in the "
+                    "provided JSON file, and an image with the given tag "
+                    "exists in the registry, the script updates the "
+                    "referenced Docker image; otherwise, it leaves the "
+                    "referenced image unchanged. The script persists the "
+                    "updated list of images in the provided output JSON file."
+                    "\n\n"
+                    "The intended use-case of this script is for a scenario "
+                    "when you know the tag used to build and push GATK-SV "
+                    "Docker images using the `build_docker.py` script, "
+                    "however, you're not sure which images are updated "
+                    "(since `build_docker.py` may rebuild and push additional"
+                    "images to the specified images, e.g., its dependencies). "
+                    "This script is predominantly developed for the Github "
+                    "actions, and is a temporary solution until the "
+                    "`build_docker.py` is re-written to provide this "
+                    "functionality.",
+        formatter_class=argparse.RawTextHelpFormatter)
+
+    parser.add_argument(
+        "dockers_json",
+        help="A JSON file containing Docker images to check. ")
```

Comment:
Agree, I set the default to `dockers.json`. I guess I am missing something in how to correctly resolve the relative path, please see the following examples. 

- The path is resolved as expected when the script is invoked from the root of the repository:

```shell
gatk_sv $ python scripts/docker/update_dockers_json.py tag

# print(os.path.join(os.path.dirname(os.path.dirname(__file__)), "input_values/dockers.json"))
# outputs: 
scripts/input_values/dockers.json
```

- The path is not resolved correctly when the script is invoked from its containing directory:

```shell
gatk_sv/scripts/docker $ python update_dockers_json.py tag
# print(os.path.join(os.path.dirname(os.path.dirname(__file__)), "input_values/dockers.json"))
# outputs: 
input_values/dockers.json
```

Since the latter better matches how it will be invoked from GitHub actions, I set the default to `../../input_values/dockers.json`. It is good enough for now, but I'm happy to update if you have a suggestion on how to improve on resolving the relative path.

Reply:
Sorry, I left out an important part of the formula, calling `os.path.abspath(__file__)`
To make it clear what's going on, I think it's helpful to separate things a little bit
```
this_script_folder = os.path.dirname(os.path.abspath(__file__))
dockers_json_path = os.path.join(os.path.dirname(this_script_folder), "input_values/dockers.json")
```

Reply:
Unfortunately, that does not work either:

```
# with this call:
gatk_sv $ python scripts/docker/update_dockers_json.py 
# The path is resolved as:
/Users/vahid/gatk_sv/scripts/input_values/dockers.json
```

Note `scripts` in the resolved path; similar issue when called from a different path.

Reply:
I guess you need one more os.path.dirname to go up one more parent directory